### ISSUE TITLE
False positive for ST11 when using a wildcard select passed to a function

### ISSUE THREAD

**Issue by reteps** [sqlfluff/sqlfluff#7234]
### Search before asking

- [x] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

Similar to #6511 , the `structure.unused_join` seems to incorrectly flag in certain cases.

```
== [/Users/peter/work/PrairieLearn/apps/prairielearn/src/ee/lib/ai-grading/ai-grading-util.sql] FAIL
L: 341 | P:  13 | ST11 | Joined table
                       | 'most_recent_instance_question_manual_grading_jobs AS
                       | mriqmgj' not referenced elsewhere in query
                       | [structure.unused_join]
```

### Expected Behaviour

A used table join does not flag `ST11`.

### Observed Behaviour

The used table join does flag `ST11`.

### How to reproduce

This query fails lint, and should not flag, even though the table is used:

```sql
SELECT
  to_jsonb(other_table.*) AS other_table_jsonb
FROM
  my_table
  LEFT JOIN other_table ON (other_table.my_table_id = my_table.id);
```

This query works:

```sql
SELECT
  to_jsonb(other_table.column) AS other_table_jsonb
FROM
  my_table
  LEFT JOIN other_table ON (other_table.my_table_id = my_table.id);
```

And this query also works:

```sql
SELECT
  other_table.*
FROM
  my_table
  LEFT JOIN other_table ON (other_table.my_table_id = my_table.id);
```

I suspect the fix for #6572 does not consider this case. cc @keraion 

### Dialect

Postgres

### Version

`sqlfluff, version 3.5.0`, `Python 3.13.5`

### Configuration

```
[sqlfluff]
templater = placeholder
dialect = postgres
large_file_skip_byte_limit = 0
sql_file_exts = .sql
exclude_rules = aliasing.column,
    aliasing.expression,
    aliasing.table,
    # False positives with to_jsonb(ALIAS)
    aliasing.unused,
    aliasing.unique.table,
    # We are fine with using SELECT * in queries.
    ambiguous.column_count,
    ambiguous.join,
    # TODO: Consider enabling this.
    ambiguous.order_by_limit,
    capitalisation.functions,
    capitalisation.identifiers,
    capitalisation.types,
    convention.casting_style,
    layout.cte_newline,
    layout.functions,
    layout.indent,
    layout.long_lines,
    layout.select_targets,
    layout.spacing,
    layout.start_of_file,
    references.consistent,
    references.from,
    references.keywords,
    references.qualification,
    structure.column_order,
    structure.distinct,
    # While explicit, this provides clarity.
    structure.else_null,
    structure.join_condition_order,
    structure.using

[sqlfluff:templater:placeholder]
param_style = dollar

# $foo is replaced by foo in the templater (no quotes), so we'll set them to something reasonable,
# like the name of the parameter in quotes, so that it can parse correctly.
start = 'start'
end = 'end'
offset = 'offset'
embedding = 'embedding'
enabled_authn_provider_ids = []
limit = 1000
```

### Are you willing to work on and submit a PR to address the issue?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)
